src/main/sensors/gyro.c

```diff
@@ -574,32 +578,45 @@ void gyroUpdateSensor(gyroSensor_t *gyroSensor)
     gyroDataAnalyse(&gyroSensor->gyroDev, gyroSensor->notchFilterDyn);
 #endif
 
-    for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-        // scale gyro output to degrees per second
-        float gyroADCf = (float)gyroSensor->gyroDev.gyroADC[axis] * gyroSensor->gyroDev.scale;
-
-#ifdef USE_GYRO_DATA_ANALYSE
-        // Apply Dynamic Notch filtering
-        if (axis == 0)
-            DEBUG_SET(DEBUG_FFT, 0, lrintf(gyroADCf)); // store raw data
-
-        if (isDynamicFilterActive())
+    if (debugMode == DEBUG_NONE) {
```

Comment:
It seems a bit unintuitive to change how filtering is done  when a user activates *any* debug mode (related to filtering or not). Wouldn't it be better to precalculate if a filtering related debug mode is active, and use the result here?

Also, on a side note, while I can see the value of what you are doing here, I also think it introduces the risk of future updates to the code only being done to one of the two code paths, in difficult do debug ways, because `debugMode` is now not free of side effects. Maybe this merits a warning in a comment?